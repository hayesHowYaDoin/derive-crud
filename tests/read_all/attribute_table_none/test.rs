use derive_crud::ReadAll;

#[derive(ReadAll)]
struct Test {
    #[crud_id]
    id: i64,
    name: String,
    location: String,
}

#[tokio::main]
async fn main() {
    let _results = Test::read_all().await;
}
