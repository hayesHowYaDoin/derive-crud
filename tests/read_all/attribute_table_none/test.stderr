error: A struct must be annotated with `#[crud_table("table_name")]` to specify the database table
 --> tests/read_all/attribute_table_none/test.rs:4:1
  |
4 | / struct Test {
5 | |     #[crud_id]
6 | |     id: i64,
7 | |     name: String,
8 | |     location: String,
9 | | }
  | |_^

error[E0599]: no function or associated item named `read_all` found for struct `Test` in the current scope
  --> tests/read_all/attribute_table_none/test.rs:13:26
   |
4  | struct Test {
   | ----------- function or associated item `read_all` not found for this struct
...
13 |     let _results = Test::read_all().await.unwrap(); // Test should fail before unwrap
   |                          ^^^^^^^^ function or associated item not found in `Test`
